packages/trpc/server/routers/viewer/teams/inviteMember/utils.ts

```diff
@@ -133,94 +167,100 @@ export function getOrgConnectionInfo({
+          verified: true,
+          invitedTo: input.teamId,
+          organizationId: orgId || null, // If the user is invited to a child team, they are automatically added to the parent org
+          teams: {
+            create: {
+              teamId: input.teamId,
+              role: input.role as MembershipRole,
+              accepted: autoAccept, // If the user is invited to a child team, they are automatically accepted
+            },
+          },
         },
-      },
-    },
-  });
+      });
 
-  // We also need to create the membership in the parent org if it exists
-  if (parentId) {
-    await prisma.membership.create({
-      data: {
-        teamId: parentId,
-        userId: createdUser.id,
-        role: input.role as MembershipRole,
-        accepted: autoAccept,
-      },
-    });
-  }
+      // We also need to create the membership in the parent org if it exists
+      if (parentId) {
+        await tx.membership.create({
+          data: {
+            teamId: parentId,
+            userId: createdUser.id,
+            role: input.role as MembershipRole,
+            accepted: autoAccept,
+          },
+        });
+      }
+    }
+  });
 }
 
-export async function createProvisionalMembership({
+export async function createProvisionalMemberships({
   input,
-  invitee,
+  invitees,
   parentId,
 }: {
   input: InviteMemberOptions["input"];
-  invitee: User;
+  invitees: User[];
   parentId?: number;
 }) {
   try {
-    await prisma.membership.create({
-      data: {
-        teamId: input.teamId,
-        userId: invitee.id,
-        role: input.role as MembershipRole,
-      },
-    });
-    // Create the membership in the parent also if it exists
-    if (parentId) {
-      await prisma.membership.create({
-        data: {
-          teamId: parentId,
+    await prisma.membership.createMany({
+      data: invitees.flatMap((invitee) => {
+        const data = [];
+        data.push({
+          teamId: input.teamId,
           userId: invitee.id,
           role: input.role as MembershipRole,
-        },
-      });
-    }
+        });
+        if (parentId) {
+          data.push({
+            teamId: parentId,
+            userId: invitee.id,
+            role: input.role as MembershipRole,
+          });
+        }
+        return data;
+      }),
+    });
   } catch (e) {
     if (e instanceof Prisma.PrismaClientKnownRequestError) {
       // Don't throw an error if the user is already a member of the team when inviting multiple users
+      console.log(Array.isArray(input.usernameOrEmail));
       if (!Array.isArray(input.usernameOrEmail) && e.code === "P2002") {
         throw new TRPCError({
           code: "FORBIDDEN",
           message: "This user is a member of this team / has a pending invitation.",
         });
       } else {
-        console.log(`User ${invitee.id} is already a member of this team.`);
+        console.log(`Trying to invite users already members of this team.`);
```

Comment:
Should we make this a error so the user can get feedback why its failing?

Reply:
I think we should just have an information message in this case, because some invites might work